### PR TITLE
Rename mongoex & add span helper

### PR DESCRIPTION
**Pull request by SemanticallyNull**

### CODE REVIEW

File: mongoex/system.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ type Config struct {
 	UseTLS  bool
 }
 
+// Load connects to mongo. The context passed in is expected to carry an o11y provider
+// and is only used for reporting (not for cancellation),
```
Comment (pete-woods) [inline mongoex/system.go:22]: In the spirit of not forcing all consumers of this library to use `system`, we should split out a `New` function, like we have for the other helper packages, that allows you just to wire in `o11y`, but not use `system`.

(non blocking for this PR, just think it's the right thing to do)

**Review by pete-woods — APPROVED** [review]
Is there a `doc.go` for this package yet?

File: README.md [hunk 2]
```diff
@@ -17,12 +17,13 @@ Browse the docs at [pkg.go.dev](https://pkg.go.dev/github.com/circleci/ex).
 - `httpserver` Starting and stopping the standard Go http server cleanly.
 - `httpserver/ginrouter` A common base for configuring a Gin router instance.
 - `httpserver/healthcheck` A healthcheck HTTP server that can accept all the checks from a `system`.
+- `mognoex` **Experimental** Common patterns using when talking to MongoDB.
```
Comment (bjohnso5) [inline README.md:20]: ~ rypo

**Comment by SemanticallyNull** [comment]
@pete-woods I've added that in with the changes you've requested. Will follow up in next PR

**Pull request merged**